entities/library.rb

```diff
@@ -0,0 +1,62 @@
+require_relative '../dependency'
+require_relative '../data/data_generator'
+
+class Library
+  include DataGenerator
+  include DataUtilities
+  include Statistics
+
+  attr_reader :authors, :books, :readers, :orders
+
+  def add_objects(*objects)
+    objects.each do |object|
+      case object.class
+      when Author
+        @authors << object
+      when Book
+        @books << object
+      when Reader
+        @readers << object
+      when Order
+        @orders << object
+      end
+    end
+  end
+
+  def generate_library
+    @authors = generate_authors
+    @books = generate_books(@authors)
+    @readers = generate_readers
+    @orders = generate_orders(@books, @readers)
+  end
+
+  def load_library_from_file(path = '')
+    data = path.empty? ? load_data : load_data(path)
+    @books = get_data(data, 'books')
```

Comment:
Maybe it would be better to save just the library object instead of saving all these entities?